app/main/autoUpdate.ts

```diff
@@ -0,0 +1,143 @@
+}
+
+async function getRemoteContent(url: string, binary = false) {
+    const options: AxiosRequestConfig = {};
+
+    if (binary) {
+        options.responseType = 'arraybuffer';
+    }
+
+    const { data } = await axios.get(url, options);
+    return data;
+}
+
+function getSha256Sum(path: string): Promise<string> {
+    const sum = createHash('sha256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => sum.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(sum.digest('hex')));
+        stream.on('error', reject);
+    });
+}
+
+function getVerifier(path: string): Promise<Verify> {
+    const verifier = createVerify('RSA-SHA256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => verifier.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(verifier));
+        stream.on('error', reject);
+    });
+}
+
+const releasesFeed = `https://github.com/${build.publish.owner}/${build.publish.repo}/releases/download`;
+
+function getVerificationFunctions(
+    { downloadedFile: filePath, releaseName, path: fileName }: RealUpdateInfo // UpdateInfo interface doesn't seem to be aligned with actual content.
+) {
+    const releaseFileUrl = `${releasesFeed}/${releaseName}/${fileName}`;
+
+    return {
+        async verifyChecksum() {
+            const remoteHash = (
+                await getRemoteContent(`${releaseFileUrl}.hash`)
+            ).trim();
+            const localHash = await getSha256Sum(filePath);
+
+            if (localHash !== remoteHash) {
+                throw new Error(
+                    `Local checksum of update does not match remote checksum (local: ${localHash}, remote: ${remoteHash}).`
+                );
+            }
+        },
+        async verifySignature() {
+            const [remoteSig, pubKey] = await Promise.all([
+                getRemoteContent(`${releaseFileUrl}.sig`, true),
+                getRemoteContent(publicKeyUrl),
+            ]);
+
+            const verifier = await getVerifier(filePath);
+            const success = verifier.verify(
+                Buffer.from(pubKey, 'ascii'),
+                remoteSig
+            );
+
+            if (!success) {
+                throw new Error('Signature verification failed.');
+            }
+        },
+    };
+}
+
+const handleUpdateDownloaded = (mainWindow: BrowserWindow) => async (
+    info: UpdateInfo
+) => {
+    mainWindow.webContents.send(updateDownloaded);
+
+    const { verifyChecksum, verifySignature } = getVerificationFunctions(
+        info as RealUpdateInfo
+    );
+
+    try {
+        await Promise.all([verifyChecksum(), verifySignature()]);
+        mainWindow.webContents.send(updateVerified);
+    } catch (e) {
+        log.error('Could not update application due to:', e);
+        mainWindow.webContents.send(updateError, 'Verification failed');
```

Comment:
After talking to Caro we came to the conclusion that this message should probably be "eased" a bit, so that the user does not panic about something having failed, as it will quite likely be because we uploaded a release incorrectly. Perhaps we can come up with something together with Jens so that the user just carries on with their work, and then they will just get the update sometime in the future.

Reply:
I agree, we've decided on "Could now apply update."